File: Dmatch-frontend/src/services/job.service.js
import axiosClient from './axiosClient';

/**
 * Lấy danh sách jobs có hỗ trợ filter + sort + pagination.
 * @param {Object} params - { keyword, location, job_type, status, job_level_id,
 *   category_ids, salary_min, salary_max, company_id, sort, page, limit }
 */
export const getJobs = (params = {}) => {
     return axiosClient.get('/api/v1/jobs', { params });
};

/**
 * Lấy chi tiết job theo ID.
 * @param {number} id
 */
export const getJobById = (id) => {
     return axiosClient.get(`/api/v1/jobs/${id}`);
};

/**
 * Lấy jobs theo company ID.
 * @param {number} companyId
 * @param {Object} params - { page, limit }
 */
export const getJobsByCompany = (companyId, params = {}) => {
     return axiosClient.get(`/api/v1/jobs/by-company/${companyId}`, { params });
};

/**
 * Tạo job mới (COMPANY/ADMIN).
 * @param {number} companyId
 * @param {Object} data - JobCreateRequest
 */
export const createJob = (companyId, data) => {
     return axiosClient.post('/api/v1/jobs', data, {
          params: { company_id: companyId },
     });
};

/**
 * Cập nhật job (COMPANY/ADMIN).
 * @param {number} jobId
 * @param {number} companyId
 * @param {Object} data - JobUpdateRequest
 */
export const updateJob = (jobId, companyId, data) => {
     return axiosClient.put(`/api/v1/jobs/${jobId}`, data, {
          params: { company_id: companyId },
     });
};

/**
 * Xóa job (COMPANY/ADMIN).
 * @param {number} jobId
 * @param {number} companyId
 */
export const deleteJob = (jobId, companyId) => {
     return axiosClient.delete(`/api/v1/jobs/${jobId}`, {
          params: { company_id: companyId },
     });
};

/**
 * Thay đổi trạng thái job (COMPANY/ADMIN).
 * @param {number} jobId
 * @param {number} companyId
 * @param {string} status
 */
export const changeJobStatus = (jobId, companyId, status) => {
     return axiosClient.put(`/api/v1/jobs/${jobId}/status`, null, {
          params: { company_id: companyId, status },
     });
};

/**
 * Lấy danh sách job levels.
 */
export const getJobLevels = () => {
     return axiosClient.get('/api/v1/jobs/levels');
};

/**
 * Lấy danh sách job categories.
 */
export const getJobCategories = () => {
     return axiosClient.get('/api/v1/jobs/categories');
};

File: job-service/src/main/java/com/dmatch/jobservice/service/interfaces/JobService.java
package com.dmatch.jobservice.service.interfaces;

import com.dmatch.jobservice.dtos.JobCategoryResponse;
import com.dmatch.jobservice.dtos.JobCreateRequest;
import com.dmatch.jobservice.dtos.JobLevelResponse;
import com.dmatch.jobservice.dtos.JobResponse;
import com.dmatch.jobservice.dtos.JobSearchRequest;
import com.dmatch.jobservice.dtos.JobUpdateRequest;
import org.springframework.data.domain.Page;

import java.util.List;
import java.util.Map;

public interface JobService {
	JobResponse createJob(JobCreateRequest request, Long companyId);
	JobResponse updateJob(Long jobId, JobUpdateRequest request, Long companyId);
	JobResponse getJobById(Long id);
	Page<JobResponse> getJobs(JobSearchRequest request, int page, int limit);
	Page<JobResponse> getJobsByCompany(Long companyId, int page, int limit);
	JobResponse changeJobStatus(Long jobId, String status, Long companyId);
	JobResponse setJobLevel(Long jobId, Long jobLevelId, Long companyId);
	JobResponse setJobCategories(Long jobId, List<Long> categoryIds, Long companyId);
	void deleteJob(Long jobId, Long companyId);

	List<JobLevelResponse> getJobLevels();
	List<JobCategoryResponse> getJobCategories();

	Map<Long, Integer> countActiveJobsByCompanyIds(List<Long> companyIds);
}
